docs/cql/dml.rst

```diff
@@ -99,11 +99,12 @@ alternatively, of the wildcard character (``*``) to select all the columns defin
 Selectors
 `````````
 
-A :token:`selector` can be one of:
+A :token:`selector` can be one of the following:
 
 - A column name of the table selected to retrieve the values for that column.
 - A casting, which allows you to convert a nested selector to a (compatible) type.
 - A function call, where the arguments are selector themselves.
+- A call to the :ref:`COUNT function <count-function>`, which counts all non-null results.
```

Comment:
I don't understand what `COUNT function <count-function>` means. What do the words "function" and "count-function" refer to?
Datastax has a better description in https://docs.datastax.com/en/cql-oss/3.3/cql/cql_reference/cqlSelect.html:
> A SELECT expression using COUNT(column_name) returns the number of non-NULL values in a column.
> A SELECT expression using COUNT(*) returns the number of rows that matched the query.

By the way, it seems this description is also missing other aggregates, like MAX!

Reply:
It is unfortunate that this makes the ".rst" file unreadable as normal text :-( Why do we need an anchor here - do you have links to it?

Anyway, instead of `A call to the :ref:`COUNT function <count-function>`, it sounds to me like a shorter and clearer phrase is just `COUNT(*)` and that's it. But note that this doesn't explain the possibility of `COUNT(column)` which also exists (and the Datastax document explains and I quoted a sentence from it).

Reply:
By the way, I took a look at the open-source Cassandra documentation https://cassandra.apache.org/doc/latest/cassandra/cql/dml.html. They have the following:

> A selector can be one of:
>
> *  A column name of the table selected, to retrieve the values for that column.
>
> *  A term, which is usually used nested inside other selectors like [functions](https://cassandra.apache.org/doc/latest/cassandra/cql/functions.html#cql-functions) (if a term is selected directly, then the corresponding column of the result-set will simply have the value of this term for every row returned).
>
 > *  A casting, which allows to convert a nested selector to a (compatible) type.
>
>  *  A function call, where the arguments are selector themselves. See the section on functions for more details.
>
>  *  The special call COUNT(*) to the [COUNT function](https://cassandra.apache.org/doc/latest/cassandra/cql/functions.html#count-function), which counts all non-null results.

Reply:
I'm starting to think that maybe the text I just quoted from Cassandra's documentation, which I said is "exactly what we need", is exactly what we did :-) Do we also have the target of "COUNT function" like Cassandra has https://cassandra.apache.org/doc/latest/cassandra/cql/functions.html#count-function and explaining all the details of "COUNT"?

Reply:
> Do we also have the target of "COUNT function" like Cassandra has https://cassandra.apache.org/doc/latest/cassandra/cql/functions.html#count-function and explaining all the details of "COUNT"?

We do, and the link using the "count-function" anchor takes you right there :) If you click "COUNT function", it opens https://docs.scylladb.com/stable/cql/functions.html#count.
I've used the term "anchor", because this is how the Sphinx links work.
- If you create a link to a page, you can simply provide the path (internal links) or URL (external links) to it. If you open such a link, you go to the top of the linked page.
- If you create a link to a section on a page, you need to a) create an anchor, b) used that anchor to create the link (i.e. you provide the anchor name, not a path or URL). See:
https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/WritingReST/Hyperlinks.html#cross-referencing
https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#cross-referencing-arbitrary-locations

In this PR, I wanted to create a link to a section - the description of the COUNT function, not the Functions page, so I had to use the anchor.

**Preview**

To preview the documentation, you need to run `make preview` as described here: https://github.com/scylladb/scylladb/blob/master/docs/README.md#building-user-documentation

We're working on adding an automatically generated preview, and we'll bring it up with the maintainers soon.

**Question**
The update I made is similar to what is in the Cassandra documentation, including the link to the details of the COUNT function.
Can I leave it as is, or do you want me to replace it with a phrase similar to what is in the Datastax docs?